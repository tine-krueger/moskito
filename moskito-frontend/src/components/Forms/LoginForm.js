import styled from 'styled-components/macro'
import ButtonBackGroup from "../Button/ButtonBackGroup"
import {useHistory} from 'react-router-dom'
import InputField from './InputField'
import {useState} from 'react'

export default function LoginForm() {
    const [ user, setUser ] = useState({
        email: '',
        password: ''
    })

    let history = useHistory()

    return (
        <LoginFormStyled onSubmit={handleSubmit}>
            <InputField type={'text'} name='email' value={user.email} onChange={handleChange} placeholder={'E-Mail'}/>
            <InputField type={'password'} name='password' value={user.password} onChange={handleChange} placeholder={'Password'}/>
            <ButtonBackGroup text1={'Login'} text2={'ZurÃ¼ck'} onClick={handleClick}/>
        </LoginFormStyled>
    )
    function handleClick() {
        history.goBack()
    }

    function handleChange(event) {
        setUser({
            ...user,
            [event.target.name]: event.target.value
        })
    }
    
    function handleSubmit(event) {
        event.preventDefault()
    }
}

const LoginFormStyled = styled.form`
    max-width: 500px;
    display: grid;
    margin:2em;
`

