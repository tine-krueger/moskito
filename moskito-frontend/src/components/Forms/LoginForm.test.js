import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import userEvent from '@testing-library/user-event'
import LoginForm from './LoginForm'
import renderWithRouter from '../../testSetup/setupTests'

describe('LoginForm', () => {

    it('renders correctly', () => {
        const { container } = render(<LoginForm/>)
        expect(container).toMatchSnapshot()
    })

    it('shows errors, if isError is truthy', () => { 
        const mockIsError = true
        const { getByText } = render(
            <LoginForm
                isError={mockIsError}
            />
        )
        expect(getByText('E-Mail oder Password-Eingabe nicht korrekt!')).toBeInTheDocument()
    })

    it('does not show errors, if isError is falsy', () => { 
        const mockIsError = false
        const { queryByText } = render(
            <LoginForm
                isError={mockIsError}
            />
        )
        expect(queryByText('E-Mail oder Password-Eingabe nicht korrekt!')).not.toBeInTheDocument()
    })

    it('redirects to find-campsite after successful login', () => { 
        const mockIsLoggedIn = true
        const { history } = renderWithRouter(
            <LoginForm
                isLoggedIn={mockIsLoggedIn}
            />
        )
        expect(history.location.pathname).toEqual('/find-campsite')
    })

    it('does not redirect to find-campsite after failed login', () => { 
        const mockIsLoggedIn = false
        const { history } = renderWithRouter(
            <LoginForm
                isLoggedIn={mockIsLoggedIn}
            />
        )
        expect(history.location.pathname).toEqual('/testroute')
    })

    it('calls userLogin with the right parameters', () => { 
        const mockUserLogin = jest.fn()
        const { getByPlaceholderText, getByText  } = render(
            <LoginForm
                userLogin={mockUserLogin}
            />
        )
        userEvent.type(getByPlaceholderText('E-Mail'), 'jane@doe.de')
        userEvent.type(getByPlaceholderText('Passwort'), '1234qwer!')
        userEvent.click(getByText('Login'))

        expect(mockUserLogin).toHaveBeenCalled()
        expect(mockUserLogin).toHaveBeenCalledWith({
            email: 'jane@doe.de',
            password: '1234qwer!'
        })
    })

    it('pushes back, when button "Zurück" is clicked', () => {
        const { history } = renderWithRouter(<LoginForm/>)
        userEvent.click(screen.getByText('Zurück'))
        expect(history.location.pathname).toEqual('/')
        expect(history.location.pathname).not.toEqual('/testroute')
    })

})