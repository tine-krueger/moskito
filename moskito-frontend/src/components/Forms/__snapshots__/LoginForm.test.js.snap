// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
.c4 {
  padding: 1em 2em;
  font-family: 'Arvo';
  font-size: 1.1rem;
  border-radius: 10px;
  width: auto;
  border: none;
  background: var(--background);
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
  color: var(--headline);
  text-transform: uppercase;
  margin: 1em auto 3em auto;
}

.c4:hover {
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
  outline: none;
}

.c3 {
  display: grid;
  justify-items: center;
}

.c3 Button {
  margin: .5em;
  width: 10em;
}

.c1 {
  grid-column: 1 / -1;
  width: 100%;
  text-align: center;
  margin: 0 auto 2em;
  margin-bottom: .5em;
}

.c1 h3 {
  text-align: left;
  padding-left: .5em;
  margin: 0 0 1em;
}

.c1 input {
  width: 100%;
  padding: 1em;
  height: 3em;
  font-size: 1em;
  border: none;
  border-radius: 10px;
  color: #6b717e;
  background-color: #d8e6e4;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c1 input:focus {
  outline: none;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3),0 0 0px 1.7px #d3a392;
}

.c1 input[type=range] {
  width: 100%;
  margin: 4.05px 0;
  background-color: transparent;
  -webkit-appearance: none;
  box-shadow: none;
}

.c1 input[type=range]:focus {
  outline: none;
  box-shadow: none;
}

.c1 input[type=range]::-webkit-slider-runnable-track {
  background: var(--background-light);
  border: 0;
  border-radius: 25px;
  width: 100%;
  height: 1.5em;
  cursor: pointer;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c1 input[type=range]::-webkit-slider-thumb {
  margin-top: -4.05px;
  width: 2em;
  height: 2em;
  background: var(--link);
  border: 0;
  border-radius: 2em;
  cursor: pointer;
  -webkit-appearance: none;
  box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
}

.c1 input[type=range]:focus::-webkit-slider-runnable-track {
  background: var(--background-light);
}

.c1 input[type=range]::-moz-range-track {
  background: var(--background-light);
  border: 0;
  border-radius: 25px;
  width: 100%;
  height: 1.5em;
  cursor: pointer;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c1 input[type=range]::-moz-range-thumb {
  width: 2em;
  height: 2em;
  background: var(--link);
  border: 0;
  border-radius: 2em;
  cursor: pointer;
  box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
}

.c1 input[type=range]::-ms-track {
  background: transparent;
  border-color: transparent;
  border-width: 9.65px 0;
  color: transparent;
  width: 100%;
  height: 1.5em;
  cursor: pointer;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c1 input[type=range]::-ms-fill-lower {
  background: var(--background-light);
  border: 0;
  border-radius: 50px;
}

.c1 input[type=range]::-ms-fill-upper {
  background: var(--background-light);
  border: 0;
  border-radius: 50px;
}

.c1 input[type=range]::-ms-thumb {
  width: 2em;
  height: 2em;
  background: var(--link);
  border: 0;
  border-radius: 2em;
  cursor: pointer;
  margin-top: 0px;
  box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
}

.c1 input[type=range]:focus::-ms-fill-lower {
  background: var(--background-light);
}

.c1 input[type=range]:focus::-ms-fill-upper {
  background: var(--background-light);
}

.c2 {
  grid-column: 1 / -1;
  width: 100%;
  text-align: center;
  margin: 0 auto 2em;
  margin-bottom: 1em;
}

.c2 h3 {
  text-align: left;
  padding-left: .5em;
  margin: 0 0 1em;
}

.c2 input {
  width: 100%;
  padding: 1em;
  height: 3em;
  font-size: 1em;
  border: none;
  border-radius: 10px;
  color: #6b717e;
  background-color: #d8e6e4;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c2 input:focus {
  outline: none;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3),0 0 0px 1.7px #d3a392;
}

.c2 input[type=range] {
  width: 100%;
  margin: 4.05px 0;
  background-color: transparent;
  -webkit-appearance: none;
  box-shadow: none;
}

.c2 input[type=range]:focus {
  outline: none;
  box-shadow: none;
}

.c2 input[type=range]::-webkit-slider-runnable-track {
  background: var(--background-light);
  border: 0;
  border-radius: 25px;
  width: 100%;
  height: 1.5em;
  cursor: pointer;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c2 input[type=range]::-webkit-slider-thumb {
  margin-top: -4.05px;
  width: 2em;
  height: 2em;
  background: var(--link);
  border: 0;
  border-radius: 2em;
  cursor: pointer;
  -webkit-appearance: none;
  box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
}

.c2 input[type=range]:focus::-webkit-slider-runnable-track {
  background: var(--background-light);
}

.c2 input[type=range]::-moz-range-track {
  background: var(--background-light);
  border: 0;
  border-radius: 25px;
  width: 100%;
  height: 1.5em;
  cursor: pointer;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c2 input[type=range]::-moz-range-thumb {
  width: 2em;
  height: 2em;
  background: var(--link);
  border: 0;
  border-radius: 2em;
  cursor: pointer;
  box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
}

.c2 input[type=range]::-ms-track {
  background: transparent;
  border-color: transparent;
  border-width: 9.65px 0;
  color: transparent;
  width: 100%;
  height: 1.5em;
  cursor: pointer;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c2 input[type=range]::-ms-fill-lower {
  background: var(--background-light);
  border: 0;
  border-radius: 50px;
}

.c2 input[type=range]::-ms-fill-upper {
  background: var(--background-light);
  border: 0;
  border-radius: 50px;
}

.c2 input[type=range]::-ms-thumb {
  width: 2em;
  height: 2em;
  background: var(--link);
  border: 0;
  border-radius: 2em;
  cursor: pointer;
  margin-top: 0px;
  box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
}

.c2 input[type=range]:focus::-ms-fill-lower {
  background: var(--background-light);
}

.c2 input[type=range]:focus::-ms-fill-upper {
  background: var(--background-light);
}

.c0 {
  max-width: 500px;
  display: grid;
  margin: 2em;
}

@supports (-ms-ime-align:auto) {
  .c1 input[type=range] {
    margin: 0;
  }
}

@supports (-ms-ime-align:auto) {
  .c2 input[type=range] {
    margin: 0;
  }
}

<div>
  <form
    class="c0"
  >
    <label
      class="c1"
      data-testid="input-field"
    >
      <h3 />
      <input
        data-testid="email"
        name="email"
        placeholder="E-Mail"
        type="text"
        value=""
      />
    </label>
    <label
      class="c2"
      data-testid="input-field"
    >
      <h3 />
      <input
        data-testid="password"
        name="password"
        placeholder="Passwort"
        type="password"
        value=""
      />
    </label>
    <div
      class="c3"
    >
      <button
        class="c4"
      >
        Login
      </button>
      <button
        class="c4"
      >
        Zur√ºck
      </button>
    </div>
  </form>
</div>
`;
